api/types/src/move_types.rs

```diff
@@ -324,6 +325,27 @@ impl MoveValue {
     }
 }
 
+impl TryFrom<MMoveValue> for MoveValue {
+    type Error = anyhow::Error;
+
+    fn try_from(val: MMoveValue) -> anyhow::Result<Self> {
+        Ok(match val {
+            MMoveValue::U8(v) => MoveValue::U8(v),
+            MMoveValue::U64(v) => MoveValue::U64(U64(v)),
+            MMoveValue::U128(v) => MoveValue::U128(U128(v)),
+            MMoveValue::Bool(v) => MoveValue::Bool(v),
+            MMoveValue::Address(v) => MoveValue::Address(v.into()),
+            MMoveValue::Vector(vals) => MoveValue::Vector(
+                vals.into_iter()
+                    .map(MoveValue::try_from)
+                    .collect::<anyhow::Result<_>>()?,
+            ),
+            MMoveValue::Signer(_) => bail!("Unexpected signer value"),
+            MMoveValue::Struct(_) => bail!("Struct conversion is currently not supported"),
```

Comment:
Having structs and specifically a vector of structs available is super important to us for more complicated usecases, which is where the view functions are really neeed :)

Reply:
Yeah, we should have struct here if we can.  I think first implementation without, then add quickly afterwards